pkg/dhclient/dhclient.go

```diff
@@ -112,7 +117,7 @@ func WriteDNSSettings(ns []net.IP, sl []string, domain string) error {
 		rc.WriteString(strings.Join(sl, " "))
 		rc.WriteString("\n")
 	}
-	return os.WriteFile("/etc/resolv.conf", rc.Bytes(), 0o644)
+	return os.WriteFile(resolvConfPath, rc.Bytes(), 0o644)
```

Comment:
would it make sense to check if this is a valid file first before writting ? if not, fall back to ResolvConfPath

Reply:
I think it is fine to leave as is. If users pass in a bad file upstream, the problem could be hard to detect on their side if we are handling for them by falling back to "etc/resolv.conf". Psuedocode wise they could do something like:

if `dhclient.WriteDNSSettings(customFilePath)` returns error, try again but passing in the defined `ResolvConfPath` -> "etc/resolv.conf" or they can choose to pass in whatever fallback file path they have.